### PR TITLE
Nfdiv 1935 add personal service reissue event

### PR DESCRIPTION
**Pull request by mapingo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-1935

### Change description ###
Add Personal Service to service method
Add Personal service to issue application service tasks
Add service selection choice to reissue event

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/IssueApplicationService.java [hunk 1]
```diff
@@ -57,11 +59,17 @@
...
-            setPostIssueState,
             details -> {
-                details.getData().getApplication().setIssueDate(LocalDate.now(clock));
+                final CaseData caseData = details.getData();
+                final Applicant applicant1 = caseData.getApplicant1();
+                final Applicant applicant2 = caseData.getApplicant2();
+                if (!applicant1.isRepresented() && !applicant2.isRepresented() && applicant2.isBasedOverseas()) {
+                    caseData.getApplication().setSolServiceMethod(PERSONAL_SERVICE);
+                }
+                caseData.getApplication().setIssueDate(LocalDate.now(clock));
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/IssueApplicationService.java:69]: Move this to two tasks SetServiceType, SetIssueDate

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/Application.java [hunk 2]
```diff
@@ -503,6 +504,11 @@ public boolean isSolicitorServiceMethod() {
         return SOLICITOR_SERVICE.equals(solServiceMethod);
     }
 
+    @JsonIgnore
+    public boolean isPersonalServiceMethod() {
+        return PERSONAL_SERVICE.equals(solServiceMethod);
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/Application.java:509]: I think we need to rename solServiceMethod, to serviceMethod

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/ServiceMethod.java [hunk 3]
```diff
@@ -14,8 +13,10 @@
     SOLICITOR_SERVICE("Solicitor Service"),
 
     @JsonProperty("courtService")
-    @JsonAlias({"personalService"})
-    COURT_SERVICE("Court Service");
+    COURT_SERVICE("Court Service"),
+
+    @JsonProperty("personalService")
+    PERSONAL_SERVICE("Personal Service");
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/ServiceMethod.java:19]: welcome back old friend

File: src/integrationTest/resources/caseworker-reissue-application-about-to-submit-app2-not-sol-rep-reissue-case-response.json [hunk 4]
```diff
@@ -65,7 +65,7 @@
     "noticeOfProceedingsEmail": "test@test.com",
     "documentsGenerated": [
       {
-        "id": "Divorce application",
+        "id": "Coversheet",
```
Comment (nitinprabhu) [inline src/integrationTest/resources/caseworker-reissue-application-about-to-submit-app2-not-sol-rep-reissue-case-response.json:68]: this is Divorce application if you check document type below

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/IssueApplicationService.java (hunk 1 above)
Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/IssueApplicationService.java]: Complete
https://github.com/hmcts/nfdiv-case-api/pull/1671/commits/e40b081b11eabec02324d47ba9208bd6282ee016

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/Application.java (hunk 2 above)
Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/Application.java]: Complete
https://github.com/hmcts/nfdiv-case-api/pull/1671/commits/a2c9da9af8fa17ba82e1aa54f58b17fdd8b18178

**Pull request closed without merging**